tests/test_aliases.py

```diff
@@ -192,3 +192,21 @@ def test_exec_alias_args(xonsh_builtins):
 
     assert stack[0][0].f_locals["myargs"] == ["arg0"]
     assert stack[0][0].f_locals["myarg0"] == "arg0"
+
+
+def test_decorator(xonsh_builtins):
+    aliases = Aliases()
+    @aliases.register
+    def debug():
+        return "DEBUG"
+
+    assert aliases['debug'] is debug
+
+
+def test_private_decorator(xonsh_builtins):
+    aliases = Aliases()
+    @aliases.register
+    def _debug():
+        return "DEBUG"
+
+    assert aliases['debug'] is _debug
```

Comment:
This seems like a good spot to use `parametrize`

Reply:
I considered using `parametrize` but I'm not sure whether it will be that useful here?

To test the `__name__` behaviour one still needs to declare the functions outside of the decorator, and the behaviour is so simple that I though it sensible just to hard code the tests. Would you prefer the two cases to be a single `test_decorator` case?